apps/web/app/modules/Courses/NewCourseView/CourseChapter.tsx

```diff
@@ -0,0 +1,139 @@
+import { CardBadge } from "~/components/CardBadge";
+import { Icon } from "~/components/Icon";
+import {
+  Accordion,
+  AccordionContent,
+  AccordionItem,
+  AccordionTrigger,
+} from "~/components/ui/accordion";
+import { Button } from "~/components/ui/button";
+import { cn, formatWithPlural } from "~/lib/utils";
+import { CourseChapterLesson } from "~/modules/Courses/NewCourseView/CourseChapterLesson";
+
+import type { GetCourseResponse } from "~/api/generated-api";
+
+const cardBadgeIcon = {
+  completed: "InputRoundedMarkerSuccess",
+  in_progress: "InProgress",
+  not_started: "NotStartedRounded",
+} as const;
+
+export const CourseChapter = ({
+  chapter,
+}: {
+  chapter: GetCourseResponse["data"]["chapters"][0];
+}) => {
+  const lessonText = formatWithPlural(chapter.lessonCount || 0, "Lesson", "Lessons");
+  const quizText = formatWithPlural(chapter.quizCount || 0, "Quiz", "Quizzes");
+
+  return (
+    <Accordion type="single" collapsible>
+      <AccordionItem value="item-1">
+        <div className="flex gap-x-4 w-full">
+          <div className="flex flex-col items-center gap-y-1">
+            <div
+              className={cn("size-10 relative rounded-full aspect-square", {
```

Comment:
We should have type for that. 
`export const ChapterProgress = {
  NOT_STARTED: "not_started",
  COMPLETED: "completed"
  IN_PROGRESS: "in_progress",
} as const;
export type ChapterProgress = (typeof ChapterProgress)[keyof typeof ChapterProgress];`

Reply:
And change it anywhere in the file

Reply:
general type not for chapters